contracts/ReputationMiningCycle.sol

```diff
@@ -50,30 +79,23 @@ contract ReputationMiningCycle {
     _;
   }
 
+  modifier challengeOpen(uint256 round, uint256 idx) {
+    // TODO: More checks that this is an appropriate time to respondToChallenge
+    require(disputeRounds[round][idx].lowerBound == disputeRounds[round][idx].upperBound);
+    _;
+  }
+
   function ReputationMiningCycle() public {
     colonyNetworkAddress = msg.sender;
     reputationMiningWindowOpenTimestamp = now;
   }
 
-  function respondToChallenge(uint256 round, uint256 idx) public {
-    // TODO: Check challenge response is valid, relating to current challenge
-    // TODO: Check challenge response relates to this hash
-
-    // Assuming that the challenge response is correct...
-    uint256 opponentIdx = (idx % 2 == 1 ? idx-1 : idx + 1);
-
-    disputeRounds[round][idx].lastResponseTimestamp = now;
-    disputeRounds[round][idx].challengeStepCompleted += 1;
-    // If our opponent responded to this challenge before we did, we should
-    // reset their 'last response' time to now, as they aren't able to respond
-    // to the next challenge before they know what it is!
-    if (disputeRounds[round][idx].challengeStepCompleted == disputeRounds[round][opponentIdx].challengeStepCompleted) {
-      disputeRounds[round][opponentIdx].lastResponseTimestamp = now;
-    }
+  function getEntryHash(address submitter, uint256 entry, bytes32 newHash) public pure returns (bytes32) {
+    return keccak256(submitter, entry, newHash);
   }
 
   function submitNewHash(bytes32 newHash, uint256 nNodes, uint256 entry) public {
```

Comment:
For clarity of this function I split the preconditions off into separate modifiers, the problem is I hit the `stack too deep` as soon as a third modifier was added so I stayed at 2 although ideally I'd have wanted a `submissionsLimitNotReached` modifier to push the following line of code to: https://github.com/JoinColony/colonyNetwork/pull/189/files?diff=unified#diff-21ee0c070df2d15abfdf2c1a0e8c8737R119

Reply:
I have no issue with this (other than my normal distaste for modifiers only used once), but reserve the right to undo this if later down the road I need to reclaim some modifiers.

I'm curious as to how you hit the limit though by messing with modifiers. Surely variables passed to modifiers are cleared off the stack once the modifier returns? Or are they treated differently from code in a function?

Reply:
I feel differently about single-use modifiers when used for the purpose of reducing complexity. This function is already ~50 lines long _with_ modifiers. I also see elegance in abstracting checks outside the core logic.

Reply:
I did not expect the introduction of modifiers to make a difference to the stack but it does. No idea why though.

Reply:
Thinking about it some more, in the compiler modifiers `_;` is replaced by the function body so they are not called like a function really which is what you are referring to.